third_party/proton/test/test_api.py

```diff
@@ -28,146 +29,147 @@ def test_profile():
-        with proton.scope("test"):
-            pass
+    temp_file = tmp_path / "test_scope.hatchet"
+    proton.start(str(temp_file).split(".")[0])
+    with proton.scope("test"):
+        pass
 
-        @proton.scope("test")
-        def foo():
-            pass
+    @proton.scope("test")
+    def foo():
+        pass
 
-        foo()
+    foo()
 
-        proton.enter_scope("test")
-        proton.exit_scope()
-        proton.finalize()
-        assert pathlib.Path(f.name).exists()
+    proton.enter_scope("test")
+    proton.exit_scope()
+    proton.finalize()
+    assert temp_file.exists()
 
 
-def test_hook():
-    with tempfile.NamedTemporaryFile(delete=True, suffix=".hatchet") as f:
-        session_id0 = proton.start(f.name.split(".")[0], hook="triton")
-        proton.activate(session_id0)
-        proton.deactivate(session_id0)
-        proton.finalize(None)
-        assert pathlib.Path(f.name).exists()
+def test_hook(tmp_path):
+    temp_file = tmp_path / "test_hook.hatchet"
+    session_id0 = proton.start(str(temp_file).split(".")[0], hook="triton")
+    proton.activate(session_id0)
+    proton.deactivate(session_id0)
+    proton.finalize(None)
+    assert temp_file.exists()
 
 
-def test_scope_metrics():
-    with tempfile.NamedTemporaryFile(delete=True, suffix=".hatchet") as f:
-        session_id = proton.start(f.name.split(".")[0])
-        # Test different scope creation methods
-        with proton.scope("test0", {"a": 1.0}):
-            pass
+def test_scope_metrics(tmp_path):
+    temp_file = tmp_path / "test_scope_metrics.hatchet"
+    session_id = proton.start(str(temp_file).split(".")[0])
+    # Test different scope creation methods
+    with proton.scope("test0", {"a": 1.0}):
+        pass
 
-        @proton.scope("test1", {"a": 1.0})
-        def foo():
-            pass
+    @proton.scope("test1", {"a": 1.0})
+    def foo():
+        pass
 
-        foo()
+    foo()
 
-        # After deactivation, the metrics should be ignored
-        proton.deactivate(session_id)
-        proton.enter_scope("test2", metrics={"a": 1.0})
-        proton.exit_scope()
+    # After deactivation, the metrics should be ignored
+    proton.deactivate(session_id)
+    proton.enter_scope("test2", metrics={"a": 1.0})
+    proton.exit_scope()
 
-        # Metrics should be recorded again after reactivation
-        proton.activate(session_id)
-        proton.enter_scope("test3", metrics={"a": 1.0})
-        proton.exit_scope()
+    # Metrics should be recorded again after reactivation
+    proton.activate(session_id)
+    proton.enter_scope("test3", metrics={"a": 1.0})
+    proton.exit_scope()
 
-        proton.enter_scope("test3", metrics={"a": 1.0})
-        proton.exit_scope()
+    proton.enter_scope("test3", metrics={"a": 1.0})
+    proton.exit_scope()
 
-        proton.finalize()
-        assert pathlib.Path(f.name).exists()
+    proton.finalize()
+    assert temp_file.exists()
+    with open(str(temp_file)) as f:
```

Comment:
NIT: Cleaner to use the pathlib open. Here and everywhere else this pattern is used.
```suggestion
    with temp_file.open() as f:
```